app/src/test/java/ru/vssemikoz/newsfeed/usacases/UpdateStorageUseCaseTest.java

```diff
@@ -66,32 +74,37 @@ private String generateRandomString(int stringSize) {
         return new String(array, Charset.forName("UTF-8"));
     }
 
-    private void initParams() {
-        filterExample = new Filter(Category.ALL, true);
-        paramsExample = new NewsFeedParams(exampleNewsList, filterExample);
+    @Test
+    public void verifyGetNewsFilteredIsCalled() {
+        when(repository.getNewsFiltered(any())).thenReturn(Single.just(exampleNewsList));
+        updateStorageUseCase.run(paramsExample);
+        verify(repository).getNewsFiltered(any());
     }
 
     @Test
-    public void verifyUpdateItemIsCalled() {
-        updateNewsItemsUseCase.run(paramsExample);
-        verify(newsStorage, atLeastOnce()).updateItem(any());
+    public void getNewsFilteredShouldContainCategory() {
+        when(repository.getNewsFiltered(any())).thenReturn(Single.just(exampleNewsList));
+        updateStorageUseCase.run(paramsExample);
+        verify(repository).getNewsFiltered(paramsCaptor.capture());
+        NewsFeedParams capturedArgument = paramsCaptor.getValue();
+        assertEquals(capturedArgument, paramsExample);
     }
 
     @Test
-    public void updateItemShouldContainNewsItemsExactlyTimes() {
-        updateNewsItemsUseCase.run(paramsExample);
-        verify(newsStorage, times(paramsExample.getNews().size())).updateItem(itemCaptor.capture());
-        List<NewsItem> capturedArgument = itemCaptor.getAllValues();
-        assertEquals(capturedArgument, exampleNewsList);
+    public void verifyInsertUniqueIsCalled() {
+        when(repository.getNewsFiltered(any())).thenReturn(Single.just(exampleNewsList));
+        updateStorageUseCase.run(any()).test();
```

Comment:
This should be checked like a regular chain using `asserValues`, `assertError`, `assertNoErrors` and `assertNoValues`.